packages/babel-types/src/definitions/es2015.js

```diff
@@ -227,6 +227,10 @@ defineType("ImportSpecifier", {
     },
     imported: {
       validate: assertNodeType("Identifier")
+    },
+    importKind: {
+      // Handle Flowtype's extension "import {typeof foo} from"
+      validate: assertOneOf(null, "type", "typeof")
```

Comment:
It would be nice if we could do extensions to nodes in corresponding files. So that flow.js could extend ImportSpecifier. But for now there is no other way afaics.